src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -570,61 +612,95 @@ private async Task ProcessSubscriptionAsync()
                 {
                     _buffer.SetLength(0);
 
-                    using (TcpConnection.DocumentDatabase.DatabaseInUse(false))
+                    var batchStatsIdPerBatch = _batchStatsId;
```

Comment:
don't do that - just use id from your object (lastbatchtats I think) - here you create copy which might be not up to date

Reply:
ok